### PR TITLE
MEP_oM: Add new fitting object

### LINKED ISSUE #1034: MEP_oM: Add generic fitting objects
#### Description:
Linear object `CableTray` has been created in oM and Engine, some thinking is needed now for its fittings. Initial thoughts is to create a placeholer object that facilitates conversion from/to Revit, having at least the following properties:

`IEnumerable Type` to hold what type of fitting it is (elbow, T, etc)
`List<ICurve> Path` to allow precise pathing through fitting to cable trays, example:
pathing1 pathing2

Any other properties we require as a minimum to make converts work? I reckon shape/material and other properties as discussed in #785 could be added afterwards.

### PR DESCRIPTION
**Pull request by tiagogrossi**
### Issues addressed by this PR

Closes #1034 

Adds a generic fitting object with basic properties and removes connection properties.

### Test files
[Here](https://burohappold.sharepoint.com/sites/BHoM/02_Current/Forms/AllItems.aspx?newTargetListUrl=%2Fsites%2FBHoM%2F02%5FCurrent&viewpath=%2Fsites%2FBHoM%2F02%5FCurrent%2FForms%2FAllItems%2Easpx&viewid=a5a93cba%2Dfcca%2D46cc%2Da31a%2D80b8d5cbfd7b&id=%2Fsites%2FBHoM%2F02%5FCurrent%2F12%5FScripts%2F02%5FPull%20Request%2FBHoM%2FBHoM%2FMEP%5FoM%2F%231034%2DAddGenericFittingObject)

### Changelog
Added `BH.oM.MEP.System.Fittings.Fitting`
Added `BH.oM.MEP.Enum.FittingType`
Modified `BH.oM.MEP.System.CableTray`
Removed `BH.oM.MEP.System.ConnectionProperties.IConnectionProperty`
Removed `BH.oM.MEP.System.ConnectionProperties.CableTrayConnectionProperty`

### CODE REVIEW

File: MEP_oM/Enums/FittingType.cs [hunk 1]
```diff
@@ -20,38 +20,27 @@
...
+        Undefined,
+        Adaptor,
+        Cap,
+        Coupling,
+        Cross,
+        Elbow,
+        Olet,
+        Plug,
+        Reducer,
+        Tee,
```
Comment (kayleighhoude) [inline MEP_oM/Enums/FittingType.cs:41]: Transition

**Review by kayleighhoude — COMMENTED** [review]
good start to fitting objects, assuming there will be a later date when we add the applicable node(s)

File: MEP_oM/Enums/FittingType.cs [hunk 2]
```diff
@@ -20,38 +20,30 @@
...
-
-        [Description("Whether the start point of the Cable Tray is connected to another segment or not.")]
-        public virtual bool IsStartConnected { get; set; }
+    /***************************************************/
 
-        [Description("Whether the end point of the Cable Tray is connected to another segment or not.")]
-        public virtual bool IsEndConnected { get; set; }       
-
-        /***************************************************/
+    [Description("A type of fitting used to transition from one segment of linear MEP object to another (cable trays, ducts or pipes)")]
```
Comment (FraserGreenroyd) [inline MEP_oM/Enums/FittingType.cs:29]: The bit in the brackets `(cable trays, ducts or pipes)` doesn't make sense when reading with the rest of the description. Is it meant to be an e.g. of linear objects? In which case it should have `e.g.` at the start, but I don't think that would enhance the description, so it's probably best to just remove it.

File: MEP_oM/Enums/FittingType.cs [hunk 3]
```diff
@@ -20,38 +20,30 @@
...
+    public enum FittingType
+    {
+        Undefined,
+        Adaptor,
```
Comment (FraserGreenroyd) [inline MEP_oM/Enums/FittingType.cs:33]:
```suggestion
        Adapter,
```

File: MEP_oM/Enums/FittingType.cs [hunk 4]
```diff
@@ -20,38 +20,30 @@
...
+    /***************************************************/
 
-        [Description("Whether the end point of the Cable Tray is connected to another segment or not.")]
-        public virtual bool IsEndConnected { get; set; }       
-
-        /***************************************************/
+    [Description("A type of fitting used to transition from one segment of linear MEP object to another (cable trays, ducts or pipes)")]
+    public enum FittingType
+    {
+        Undefined,
```
Comment (FraserGreenroyd) [inline MEP_oM/Enums/FittingType.cs:32]: A few of these enum options could do with `description` attributes to help boost their understandability to non-expert users.

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]
See comments on files for more details.

The deletion of the property from `CableTray` will require versioning support.

**Comment by tiagogrossi** [comment]
@BHoMBot check-versioning

**Comment by tiagogrossi** [comment]
@BHoMBot check versioning

**Comment by tiagogrossi** [comment]
@BHoMBot check required

**Comment by michaelhoehn** [comment]
@BHoMBot check core

**Comment by michaelhoehn** [comment]
@BHoMBot check installer

**Comment by michaelhoehn** [comment]
@BHoMBot check versioning

**Comment by michaelhoehn** [comment]
@BHoMBot check compliance

**Comment by michaelhoehn** [comment]
@BHoMBot check serialisation

**Review by FraserGreenroyd — APPROVED** [review]
Reviewed through with @michaelhoehn and all is good now.

**Comment by FraserGreenroyd** [comment]
@BHoMBot check ready-to-merge

**Pull request merged**